src/main.rs

```diff
@@ -205,11 +203,17 @@ Report issues to <https://github.com/lunaryorn/mdcat>.",
 
     let matches = app.get_matches();
     let arguments = Arguments::from_matches(&matches).unwrap_or_else(|e| e.exit());
-    match process_arguments(size, arguments) {
-        Ok(_) => std::process::exit(0),
-        Err(error) => {
-            eprintln!("Error: {}", error);
-            std::process::exit(1);
+
+    if arguments.detect_only {
```

Comment:
As said above, please move this branch back into `process_arguments`.

Reply:
done